plugin/storage/grpc/shared/streaming_writer.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+var (
+	_ spanstore.Writer = (*streamingSpanWriter)(nil)
+)
+
+// streamingSpanWriter wraps storage_v1.StreamingSpanWriterPluginClient into spanstore.Writer
+type streamingSpanWriter struct {
+	client storage_v1.StreamingSpanWriterPluginClient
+	stream storage_v1.StreamingSpanWriterPlugin_WriteSpanStreamClient
+}
+
+// WriteSpan write span into stream
+func (s *streamingSpanWriter) WriteSpan(ctx context.Context, span *model.Span) error {
+	if s.stream == nil {
```

Comment:
Is it possible that this method will be called concurrently? A general SpanWriter is expected to be concurrent because Collector uses the same storage from multiple workers. I don't think there's any synchronization point from top-level SpanWriter to this class.

Reply:
Yes, `stream.Send` can't be called concurrently too, let me modify it